File: Nx3CS/bin/test/src/nx3/PStaveRectCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PStaveRectCalculator : global::haxe.lang.HxObject {
		public    PStaveRectCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PStaveRectCalculator(global::nx3.PNote note){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				global::nx3.PStaveRectCalculator.__hx_ctor_nx3_PStaveRectCalculator(this, note);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PStaveRectCalculator(global::nx3.PStaveRectCalculator __temp_me118, global::nx3.PNote note){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				__temp_me118.note = note;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				return new global::nx3.PStaveRectCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				return new global::nx3.PStaveRectCalculator(((global::nx3.PNote) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PNote note;
		
		public virtual   global::nx3.geom.Rectangle getStaveRect(){
			unchecked {
				#line 22 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				if ( ! (string.Equals(global::Type.enumConstructor(this.note.nnote.type), "Note")) ) {
					#line 22 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					return default(global::nx3.geom.Rectangle);
				}
				
				#line 23 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				if (( global::nx3.ENoteValTools.stavinglevel(this.note.nnote.@value) < 1 )) {
					#line 23 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					return default(global::nx3.geom.Rectangle);
				}
				
				#line 25 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				double headw = default(double);
				#line 25 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				{
					#line 25 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					global::nx3.EHeadValueType _g = global::nx3.ENoteValTools.head(this.note.nnote.@value);
					#line 25 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					switch (global::Type.enumIndex(_g)){
						case 2:
						{
							#line 27 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							headw = 2.2;
							#line 27 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							break;
						}
						
						
						default:
						{
							#line 28 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							headw = 1.6;
							#line 28 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							break;
						}
						
					}
					
				}
				
				#line 30 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				global::nx3.geom.Rectangle rect = default(global::nx3.geom.Rectangle);
				if (( this.note.getDirection() == global::nx3.EDirectionUD.Up )) {
					#line 33 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					rect = new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(( this.note.nnote.getBottomLevel() - ((double) (7) ) ), true), new global::haxe.lang.Null<double>(headw, true), new global::haxe.lang.Null<double>(((double) (7) ), true));
				}
				 else {
					#line 37 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					rect = new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>( - (headw) , true), new global::haxe.lang.Null<double>(((double) (this.note.nnote.getTopLevel()) ), true), new global::haxe.lang.Null<double>(headw, true), new global::haxe.lang.Null<double>(((double) (7) ), true));
				}
				
				#line 39 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				rect.offset(this.note.getXOffset(), ((double) (0) ));
				return rect;
			}
			#line default
		}
		
		
		public virtual   global::nx3.geom.Rectangle getFlagRect(){
			unchecked {
				#line 45 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				if ( ! (string.Equals(global::Type.enumConstructor(this.note.nnote.type), "Note")) ) {
					#line 45 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					return default(global::nx3.geom.Rectangle);
				}
				
				#line 46 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				if (( global::nx3.ENoteValTools.beaminglevel(this.note.nnote.@value) < 1 )) {
					#line 46 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					return default(global::nx3.geom.Rectangle);
				}
				
				#line 47 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				global::nx3.PBeamgroup beamgroup = this.note.getBeamgroup();
				#line 50 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				if (( ( beamgroup != default(global::nx3.PBeamgroup) ) && ( beamgroup.pnotes.length == 1 ) )) {
					#line 52 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					if (( global::nx3.ENoteValTools.beaminglevel(this.note.nnote.@value) > 0 )) {
						#line 54 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						double headw = default(double);
						#line 54 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						{
							#line 54 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							global::nx3.EHeadValueType _g = global::nx3.ENoteValTools.head(this.note.nnote.@value);
							#line 54 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							switch (global::Type.enumIndex(_g)){
								case 2:
								{
									#line 56 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
									headw = 2.2;
									#line 56 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
									break;
								}
								
								
								default:
								{
									#line 57 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
									headw = 1.6;
									#line 57 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
									break;
								}
								
							}
							
						}
						
						#line 59 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						global::nx3.geom.Rectangle rect = default(global::nx3.geom.Rectangle);
						if (( this.note.getDirection() == global::nx3.EDirectionUD.Up )) {
							#line 62 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							rect = new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(headw, true), new global::haxe.lang.Null<double>(( this.note.nnote.getBottomLevel() - ((double) (7) ) ), true), new global::haxe.lang.Null<double>(2.6, true), new global::haxe.lang.Null<double>(4.8, true));
						}
						 else {
							#line 66 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
							rect = new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>( - (headw) , true), new global::haxe.lang.Null<double>(( ( this.note.nnote.getTopLevel() + ((double) (7) ) ) - 4.8 ), true), new global::haxe.lang.Null<double>(2.6, true), new global::haxe.lang.Null<double>(4.8, true));
						}
						
						#line 68 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						rect.offset(this.note.getXOffset(), ((double) (0) ));
						return rect;
					}
					
				}
				
				#line 72 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				return default(global::nx3.geom.Rectangle);
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				switch (hash){
					case 1225398258:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						this.note = ((global::nx3.PNote) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				switch (hash){
					case 1216318886:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getFlagRect") ), ((int) (1216318886) ))) );
					}
					
					
					case 173809949:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getStaveRect") ), ((int) (173809949) ))) );
					}
					
					
					case 1225398258:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return this.note;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				switch (hash){
					case 1216318886:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return this.getFlagRect();
					}
					
					
					case 173809949:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return this.getStaveRect();
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				baseArr.push("note");
				#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
				{
					#line 11 "F:\\nx3\\src\\nx3\\PStaveRectCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PStaveRectCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PStaveRectCalculator extends haxe.lang.HxObject
{
	public    PStaveRectCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PStaveRectCalculator(nx3.PNote note)
	{
		nx3.PStaveRectCalculator.__hx_ctor_nx3_PStaveRectCalculator(this, note);
	}
	
	
	public static   void __hx_ctor_nx3_PStaveRectCalculator(nx3.PStaveRectCalculator __temp_me135, nx3.PNote note)
	{
		__temp_me135.note = note;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PStaveRectCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PStaveRectCalculator(((nx3.PNote) (arr.__get(0)) ));
	}
	
	
	public  nx3.PNote note;
	
	public   nx3.geom.Rectangle getStaveRect()
	{
		if ( ! (haxe.lang.Runtime.valEq(haxe.root.Type.enumConstructor(this.note.nnote.type), "Note")) ) 
		{
			return null;
		}
		
		if (( nx3.ENoteValTools.stavinglevel(this.note.nnote.value) < 1 )) 
		{
			return null;
		}
		
		double headw = 0.0;
		{
			nx3.EHeadValueType _g = nx3.ENoteValTools.head(this.note.nnote.value);
			switch (haxe.root.Type.enumIndex(_g))
			{
				case 2:
				{
					headw = 2.2;
					break;
				}
				
				
				default:
				{
					headw = 1.6;
					break;
				}
				
			}
			
		}
		
		nx3.geom.Rectangle rect = null;
		if (( this.note.getDirection() == nx3.EDirectionUD.Up )) 
		{
			rect = new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (( this.note.nnote.getBottomLevel() - ((double) (7) ) )) ), ((java.lang.Object) (headw) ), ((double) (7) ));
		}
		 else 
		{
			rect = new nx3.geom.Rectangle(((java.lang.Object) ( - (headw) ) ), ((java.lang.Object) (this.note.nnote.getTopLevel()) ), ((java.lang.Object) (headw) ), ((double) (7) ));
		}
		
		rect.offset(this.note.getXOffset(), ((double) (0) ));
		return rect;
	}
	
	
	public   nx3.geom.Rectangle getFlagRect()
	{
		if ( ! (haxe.lang.Runtime.valEq(haxe.root.Type.enumConstructor(this.note.nnote.type), "Note")) ) 
		{
			return null;
		}
		
		if (( nx3.ENoteValTools.beaminglevel(this.note.nnote.value) < 1 )) 
		{
			return null;
		}
		
		nx3.PBeamgroup beamgroup = this.note.getBeamgroup();
		if (( ( beamgroup != null ) && ( beamgroup.pnotes.length == 1 ) )) 
		{
			if (( nx3.ENoteValTools.beaminglevel(this.note.nnote.value) > 0 )) 
			{
				double headw = 0.0;
				{
					nx3.EHeadValueType _g = nx3.ENoteValTools.head(this.note.nnote.value);
					switch (haxe.root.Type.enumIndex(_g))
					{
						case 2:
						{
							headw = 2.2;
							break;
						}
						
						
						default:
						{
							headw = 1.6;
							break;
						}
						
					}
					
				}
				
				nx3.geom.Rectangle rect = null;
				if (( this.note.getDirection() == nx3.EDirectionUD.Up )) 
				{
					rect = new nx3.geom.Rectangle(((java.lang.Object) (headw) ), ((java.lang.Object) (( this.note.nnote.getBottomLevel() - ((double) (7) ) )) ), ((java.lang.Object) (2.6) ), ((java.lang.Object) (4.8) ));
				}
				 else 
				{
					rect = new nx3.geom.Rectangle(((java.lang.Object) ( - (headw) ) ), ((java.lang.Object) (( ( this.note.nnote.getTopLevel() + ((double) (7) ) ) - 4.8 )) ), ((java.lang.Object) (2.6) ), ((java.lang.Object) (4.8) ));
				}
				
				rect.offset(this.note.getXOffset(), ((double) (0) ));
				return rect;
			}
			
		}
		
		return null;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef764 = true;
			switch (field.hashCode())
			{
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef764 = false;
						this.note = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef764) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef765 = true;
			switch (field.hashCode())
			{
				case -603228954:
				{
					if (field.equals("getFlagRect")) 
					{
						__temp_executeDef765 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getFlagRect"))) );
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef765 = false;
						return this.note;
					}
					
					break;
				}
				
				
				case -2038299491:
				{
					if (field.equals("getStaveRect")) 
					{
						__temp_executeDef765 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getStaveRect"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef765) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef766 = true;
			switch (field.hashCode())
			{
				case -603228954:
				{
					if (field.equals("getFlagRect")) 
					{
						__temp_executeDef766 = false;
						return this.getFlagRect();
					}
					
					break;
				}
				
				
				case -2038299491:
				{
					if (field.equals("getStaveRect")) 
					{
						__temp_executeDef766 = false;
						return this.getStaveRect();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef766) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("note");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


